modules/javafx.controls/src/test/java/test/javafx/scene/control/SelectionFocusModelMemoryTest.java

```diff
@@ -0,0 +1,298 @@
+        ObservableList<String> data = FXCollections.observableArrayList("Apple", "Orange", "Banana");
+        data.forEach(text -> root.getChildren().add(new TreeItem<>(text)));
+        TreeView<String> control = new TreeView<>(root);
+        WeakReference<SelectionModel<?>> weakRef = new WeakReference<>(control.getSelectionModel());
+        MultipleSelectionModel<TreeItem<String>> replacingSm = TreeViewShim.get_TreeViewBitSetSelectionModel(control);
+        maybeShowControl(control);
+        control.setSelectionModel(replacingSm);
+        attemptGC(weakRef, 10);
+        assertNull("selectionModel must be gc'ed", weakRef.get());
+    }
+
+    @Test
+    public void testTreeTableViewSelectionModel() {
+        TreeItem<String> root = new TreeItem<>("root");
+        ObservableList<String> data = FXCollections.observableArrayList("Apple", "Orange", "Banana");
+        data.forEach(text -> root.getChildren().add(new TreeItem<>(text)));
+        TreeTableView<String> control = new TreeTableView<>(root);
+        WeakReference<SelectionModel<?>> weakRef = new WeakReference<>(control.getSelectionModel());
+        TreeTableViewSelectionModel<String> replacingSm = (TreeTableViewSelectionModel<String>) TreeTableViewShim.get_TreeTableViewArrayListSelectionModel(control);
+        maybeShowControl(control);
+        control.setSelectionModel(replacingSm);
+        attemptGC(weakRef, 10);
+        assertNull("selectionModel must be gc'ed", weakRef.get());
+    }
+
+    @Test
+    public void testTableViewSelectionModel() {
+        TableView<String> control = new TableView<>(FXCollections.observableArrayList("Apple", "Orange", "Banana"));
+        WeakReference<SelectionModel<?>> weakRef = new WeakReference<>(control.getSelectionModel());
+        TableViewSelectionModel<String> replacingSm = TableViewShim.get_TableViewArrayListSelectionModel(control);
+        maybeShowControl(control);
+        control.setSelectionModel(replacingSm);
+        attemptGC(weakRef, 10);
+        assertNull("selectionModel must be gc'ed", weakRef.get());
+    }
+
+    @Test
+    public void testListViewSelectionModel() {
+        ListView<String> control = new ListView<>(FXCollections.observableArrayList("Apple", "Orange", "Banana"));
+        WeakReference<SelectionModel<?>> weakRef = new WeakReference<>(control.getSelectionModel());
+        MultipleSelectionModel<String> replacingSm = ListViewShim.getListViewBitSetSelectionModel(control);
+        maybeShowControl(control);
+        control.setSelectionModel(replacingSm);
+        attemptGC(weakRef, 10);
+        assertNull("selectionModel must be gc'ed", weakRef.get());
+    }
+
+    @Test
+    public void testTabPaneSelectionModel() {
+        // FIXME
+        // can't formally ignore just one parameter, so backing out if showBeforeReplaceSM
+        // has its own issue https://bugs.openjdk.java.net/browse/JDK-8241737
+        if (showBeforeReplaceSM) return;
```

Comment:
@Ignore is used to ignore the test that are known to fail, It also becomes easy to track the ignored test with the @Ignore tag. I think it will be good to tag this scenario with a commented @Ignore inside the if statement
`//@Ignore("JDK-8241737")`